deeprankcore/dataset.py

```diff
@@ -685,7 +693,30 @@ def __init__( # pylint: disable=too-many-arguments, too-many-locals
                 dataset_train._compute_mean_std()
             self.means = dataset_train.means
             self.devs = dataset_train.devs
-
+            
+    def _check_inherited_param(
+        self,
+        inherited_param: List[str] = None,
```

Comment:
inherited_param and dataset_train shouldn't be optional, since the check function is called only when you actually have both inherited_param and dataset_train assigned